metagraph/core/plugin.py

```diff
@@ -235,9 +252,17 @@ def compute_abstract_properties(cls, obj, props: Set[str]) -> Dict[str, Any]:
 
         The properties are cached to speed up future calls for the same properties.
         """
+        assert cls.is_typeclass_of(
+            obj
+        ), f"Cannot compute abstract properties of {obj} using {cls}"
+
         if len(props) == 0:
             return {}
 
+        # Be forgiving if a single string is passed in
```

Comment:
Since `compute_abstract_properties` handles `props` being a string, it would be nice to change the type hint from `props: Set[str]` to `props: Union[str, Set[str]]`.